xbmc/windows/GUIWindowLoginScreen.cpp

```diff
@@ -338,11 +338,10 @@ void CGUIWindowLoginScreen::LoadProfile(unsigned int profile)
 
   JSONRPC::CJSONRPC::Initialize();
 
-  // Restart context menu manager
-  CServiceBroker::GetContextMenuManager().Init();
-
-  // restart PVR services
-  CServiceBroker::GetPVRManager().Init();
+  if (!g_application.m_ServiceManager->InitStageThree())
```

Comment:
As far as I can see IniStageThree calls `CPeripherals::Initialise()`, but there seems to be no call to `CPeripherals::Clear()` (which is the counterpart to Initialise() - see DeinitStageThree). No good, if you ask me...